### ISSUE TITLE
How to test asciidoc syntax/output

### ISSUE THREAD

**Issue by jasonrhodes** [elastic/docs#415]
**tl;dr: how do I reliably test the output of the asciidoc generated HTML before merging docs changes?**

Hello, I've been trying to fix docs seen here: https://www.elastic.co/guide/en/kibana/master/apm-settings-kb.html

I edited the file in the kibana repo (https://github.com/elastic/kibana/edit/master/docs/settings/apm-settings.asciidoc) in order to figure out how to display the `*` character twice in one line without triggering "bold". We originally had encoded characters which were showing up wrong, then when I switched to escaping the first `*` in the line with a `\`, GitHub showed this correctly as did asciidoclive.com, but when it finally built and was deployed, the `\*` shows up on the page.

I pulled this repo so I could build the docs locally and see how our asciidoc parsing is working (since it's obviously different from how GitHub and asciidoclive.com are parsing it). But when I run the build_docs.pl command, I get linting errors:

```
$ ./build_docs.pl --doc ../kibana/docs/settings/apm-settings.asciidoc
Building HTML from ../kibana/docs/settings/apm-settings.asciidoc

Error executing: a2x -v --icons -d book -f chunked -a showcomments=1 -a lang=en -a base_edit_url= -a root_dir=/Users/jason/o_O/docs/../kibana/docs/settings --xsl-file resources/website_chunked.xsl --asciidoc-opts -fresources/es-asciidoc.conf --destination-dir=/Users/jason/o_O/docs/html_docs --xsltproc-opts --stringparam toc.max.depth '5' --xsltproc-opts --stringparam toc.section.depth '0' --xsltproc-opts --stringparam chunk.section.depth '0' --xsltproc-opts --stringparam local.book.version 'test build' --xsltproc-opts --stringparam local.book.multi_version '0' --xsltproc-opts --stringparam local.page.header '' --xsltproc-opts --stringparam local.book.section.title 'Learn/Docs/' --xsltproc-opts --stringparam local.book.subject '' --xsltproc-opts --stringparam local.noindex '' /Users/jason/o_O/docs/../kibana/docs/settings/apm-settings.asciidoc

asciidoc: WARNING: apm-settings.asciidoc: line 3: section title out of sequence: expected level 1, got level 2


/Users/jason/o_O/docs/html_docs/apm-settings.xml:87: element section: validity error : Element section content does not follow the DTD, expecting (sectioninfo? , (title , subtitle? , titleabbrev?) , (toc | lot | index | glossary | bibliography)* , (((calloutlist | glosslist | bibliolist | itemizedlist | orderedlist | segmentedlist | simplelist | variablelist | caution | important | note | tip | warning | literallayout | programlisting | programlistingco | screen | screenco | screenshot | synopsis | cmdsynopsis | funcsynopsis | classsynopsis | fieldsynopsis | constructorsynopsis | destructorsynopsis | methodsynopsis | formalpara | para | simpara | address | blockquote | graphic | graphicco | mediaobject | mediaobjectco | informalequation | informalexample | informalfigure | informaltable | equation | example | figure | table | msgset | procedure | sidebar | qandaset | task | anchor | bridgehead | remark | highlights | abstract | authorblurb | epigraph | indexterm | beginpage)+ , (refentry* | section* | simplesect*)) | refentry+ | section+ | simplesect+) , (toc | lot | index | glossary | bibliography)*), got (title title simpara bridgehead variablelist )
</section>
          ^
/Users/jason/o_O/docs/html_docs/apm-settings.xml:88: element book: validity error : Element book content does not follow the DTD, expecting ((title , subtitle? , titleabbrev?)? , bookinfo? , (dedication | toc | lot | glossary | bibliography | preface | chapter | reference | part | article | appendix | index | setindex | colophon)*), got (bookinfo section )
</book>
       ^


---
a2x: ERROR: "xmllint" --nonet --noout --valid "/Users/jason/o_O/docs/html_docs/apm-settings.xml" returned non-zero exit status 4
``` 

**So my main question is: how do I reliably test the output of the asciidoc generated HTML before merging docs changes?**

**Comment by lcawl** [comment]
I typically just build the book that contains the changed file. The build commands for each book can be found here: https://github.com/elastic/docs/blob/master/doc_build_aliases.sh
Since I run them many times a day, I added that file to my bash_profile (as described at top of file) and can just use the "docbldkb" alias.  That book in particular is a quick build.

**Comment by jasonrhodes** [comment]
@lcawl thanks, I think that worked for me to test locally ... I'd still like to understand what we are using to parse/compile asciidoc so I can figure out if there is a syntax reference we can use without trial and error builds. Do you know the answer to that?
